src/current/v25.3/ldap-authentication.md

```diff
@@ -161,12 +192,139 @@ cockroach sql --url "postgresql://username:ldap_password@host:26257" --certs-dir
+
+### Viewing provisioned users
+
+Auto-provisioned users can be identified by their `PROVISIONSRC` role option:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+-- View all auto-provisioned users (users with PROVISIONSRC role option)
+SELECT * FROM [SHOW USERS] AS u 
+WHERE EXISTS (
+  SELECT 1 FROM unnest(u.options) AS opt 
+  WHERE opt LIKE 'PROVISIONSRC=ldap:%'
+);
+~~~
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+-- View all manually created users (users without PROVISIONSRC role option)
+SELECT * FROM [SHOW USERS] AS u 
+WHERE NOT EXISTS (
+  SELECT 1 FROM unnest(u.options) AS opt 
+  WHERE opt LIKE 'PROVISIONSRC=ldap:%'
+);
+~~~
+
+### Cleaning up users removed from Active Directory
+
+Auto-provisioned users who have been removed or deactivated in Active Directory will not be automatically removed from CockroachDB. To identify and clean up these orphaned accounts:
+
+**Step 1: Export auto-provisioned users from CockroachDB**
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+-- Export list of auto-provisioned usernames for comparison with Active Directory
+SELECT u.username FROM [SHOW USERS] AS u 
+WHERE EXISTS (
+  SELECT 1 FROM unnest(u.options) AS opt 
+  WHERE opt LIKE 'PROVISIONSRC=ldap:%'
+);
+~~~
+
+**Step 2: Cross-reference with Active Directory**
+
+Use your organization's directory tools to verify which of these users still exist in Active Directory. For example, you might use PowerShell with Active Directory cmdlets:
+
+```powershell
+# Example PowerShell script to check if users exist in AD
+$cockroachUsers = @("user1", "user2", "user3")  # Replace with actual usernames
+$orphanedUsers = @()
+
+foreach ($user in $cockroachUsers) {
+    try {
+        Get-ADUser -Identity $user -ErrorAction Stop | Out-Null
+    } catch {
+        $orphanedUsers += $user
+        Write-Host "User not found in AD: $user"
+    }
+}
+```
+
+**Step 3: Remove orphaned users**
+
+Before dropping users confirmed to no longer exist in Active Directory, check for any privileges that were granted directly to the user:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+-- Check for direct grants to the user (privileges inherited through roles won't block DROP USER)  
+SHOW GRANTS FOR username;
+~~~
+
+If any direct grants exist, revoke them before dropping the user. For users confirmed to no longer exist in Active Directory:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+-- Remove users that no longer exist in Active Directory
+DROP USER username1, username2, username3;
+~~~
+
+{{site.data.alerts.callout_info}}
+Users cannot be dropped if they have direct privilege grants or own database objects. For complete requirements, see [`DROP USER`]({% link {{ page.version.version }}/drop-user.md %}). When using both automatic user provisioning and LDAP Authorization, consider granting privileges primarily through roles (mapped to AD groups) rather than directly to users to simplify cleanup operations.
```

Comment:
```suggestion
Users cannot be dropped if they have direct privilege grants or own database objects. For complete requirements, refer to [`DROP USER`]({% link {{ page.version.version }}/drop-user.md %}). When using both automatic user provisioning and LDAP Authorization, consider granting privileges primarily through roles (mapped to AD groups) rather than directly to users to simplify cleanup operations.
```